### PR TITLE
Refactored the JsonApiSerializer

### PR DESCRIPTION
**Pull request by vanbrabantf**
### Changed
  - Divided the methods into smaller chunks for better readability.

### CODE REVIEW

**Comment by willishq** [comment]
Hi, I'd be happy to merge this in if you can change the method names to adhere to PSR-2!

https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md

**Comment by hannesvdvreken** [comment]
I'd make all internal method private if they are not used by child classes. This gives you the option to change them later on without worrying about BC.

**Comment by hannesvdvreken** [comment]
:clap: you're first open source contribution, @vanbrabantf. Congrats!

**Comment by vanbrabantf** [comment]
dope, thanks guys!

File: src/Serializer/JsonApiSerializer.php [hunk 1]
```diff
@@ -454,4 +407,152 @@ protected function shouldIncludeLinks()
...
+
+
+    /**
+     * @param $data
+     * @param $relationship
+     * @param $key
+     *
+     * @return array
+     */
+    private function FillRelationshipAsSingleResource($data, $relationship, $key)
```
Comment (webda2l) [inline src/Serializer/JsonApiSerializer.php]: fillRelationshipAsSingleResource

**Pull request merged**